nucypher/blockchain/eth/registry.py

```diff
@@ -231,33 +188,93 @@ def search(self, contract_name: str=None, contract_address: str=None):
         return contracts if contract_name else contracts[0]
 
 
-class LocalEthereumContractRegistry(EthereumContractRegistry):
+class LocalContractRegistry(BaseContractRegistry):
+
+    def __init__(self, filepath: str, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__filepath = filepath
+        self.log.info("Using contract registry {}".format(self.__filepath))
+
+    def __repr__(self):
+        r = f"{self.__class__.__name__}(filepath={self.filepath})"
+        return r
+
+    @property
+    def filepath(self) -> str:
```

Comment:
I see that we use both `self.__filepath` or `self.filepath`. Which one do we actually want to use (and why have the `@property` at all - to prevent accidental changing?)

Reply:
Yes.